python/ray/air/config.py

```diff
@@ -273,6 +273,58 @@ class FailureConfig:
     max_failures: int = 0
 
 
+@dataclass
+@PublicAPI(stability="alpha")
+class CheckpointConfig:
```

Comment:
Why are we introducing a new concept here? Shouldn't we just use `CheckpointStrategy`?

Reply:
Can we use checkpoint strategy with tune.run? I was under the impression it was an internal concept